### PR TITLE
Improve UT coverage in cloud/scope package for VPC

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Improve UT coverage in cloud/scope package for VPC

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Improve UT coverage in cloud/scope package for VPC
```

### CODE REVIEW

**Comment by mkumatag** [comment]
@Prajyot-Parab can you resolve the merge conflict?

**Review by Amulyam24 — APPROVED** [review]
/lgtm

File: cloud/scope/cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+					Zone:          "foo-zone-1",
+				}}
+			expectedOutput = &vpcv1.VPC{
+				Name: core.StringPtr("foo-vpc-1"),
+				ID:   core.StringPtr("foo-vpc-1-id"),
+				DefaultSecurityGroup: &vpcv1.SecurityGroupReference{
+					ID: core.StringPtr("foo-security-group-1"),
+				}}
+
+			scope.IBMVPCCluster.Spec = vpcClusterCustom.Spec
```
Comment (mkumatag) [inline cloud/scope/cluster_test.go:159]: can we assign the content `vpcClusterCustom.spec` to here directly?

File: cloud/scope/common.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (mkumatag) [inline cloud/scope/common.go:1]: if this is used only for the tests, can be renamed to `common_test.go` or something else to restrict these functions only to the tests

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2/klogr"
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/vpc/mock"
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/cluster_test.go]: done.

File: cloud/scope/suite_test.go [hunk 4]
```diff
@@ -16,51 +16,80 @@ limitations under the License.
...
-	"github.com/onsi/ginkgo/reporters"
+	"github.com/golang/mock/gomock"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/kubernetes/scheme"
-	"k8s.io/client-go/rest"
-	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	ctrl "sigs.k8s.io/controller-runtime"
 
+	// +kubebuilder:scaffold:imports
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/suite_test.go]: Not needed, fixed.

File: cloud/scope/cluster_test.go [hunk 5]
```diff
@@ -0,0 +1,555 @@
...
+				IBMVPCCluster: newVPCCluster(clusterName),
+				Client:        testEnv.Client,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := NewClusterScope(tc.params)
+			g.Expect(err).To(Not(BeNil()))
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/cluster_test.go:89]: done.

File: cloud/scope/common.go (hunk 2 above)
Comment (Prajyot-Parab) [reply cloud/scope/common.go]: Renamed.

File: cloud/scope/cluster_test.go [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+package scope
+
+import (
+	"testing"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/cluster_test.go:26]: Yes both being used.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/cluster_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Prajyot-Parab) [reply cloud/scope/cluster_test.go]: done.

File: cloud/scope/machine_test.go [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```
Comment (mkumatag) [inline cloud/scope/machine_test.go:27]: fix it here as well

File: cloud/scope/machine_test.go [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2/klogr"
+	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/vpc/mock"
```
Comment (mkumatag) [inline cloud/scope/machine_test.go:34]: create different section for these 2 imports as well.

File: cloud/scope/suite_test.go [hunk 9]
```diff
@@ -16,51 +16,79 @@ limitations under the License.
...
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	infrav1beta1 "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta1"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/vpc/mock"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/test/helpers"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
```
Comment (mkumatag) [inline cloud/scope/suite_test.go:31]: this import should be in the above section with other imports

**Review by mkumatag — CHANGES_REQUESTED** [review]
some more nits

File: cloud/scope/suite_test.go (hunk 9 above)
Comment (Prajyot-Parab) [reply cloud/scope/suite_test.go]: Fixed.

File: cloud/scope/machine_test.go (hunk 8 above)
Comment (Prajyot-Parab) [reply cloud/scope/machine_test.go]: Fixed.

File: cloud/scope/machine_test.go (hunk 7 above)
Comment (Prajyot-Parab) [reply cloud/scope/machine_test.go]: Fixed.

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**